plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -146,6 +147,89 @@ fu_plugin_thunderbolt_is_host (GUdevDevice *device)
 	return g_str_has_prefix (name, "domain");
 }
 
+static gchar *
+fu_plugin_thunderbolt_get_force_power_path (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+
+	GList *devices;
+	const gchar *basepath;
+	g_autofree gchar *built_path = NULL;
+
+	devices = g_udev_client_query_by_subsystem (data->udev, "wmi");
+	for (GList *l = devices; l != NULL; l = l->next) {
+		GUdevDevice *device = l->data;
+		basepath = g_udev_device_get_sysfs_path (device);
+		built_path = g_build_path ("/", basepath,
+					   "force_power", NULL);
+		if (g_file_test (built_path, G_FILE_TEST_IS_REGULAR))
+			break;
+		built_path = NULL;
+	}
+	g_list_foreach (devices, (GFunc) g_object_unref, NULL);
+	g_list_free (devices);
+
+	if (built_path)
+		return g_strdup (built_path);
```

Comment:
g_steal_pointer() maybe?

Reply:
IIUC, we don't need the `if` in this case

Reply:
Yep, using that thanks!